service/lib/agama/storage/bootloader.rb

```diff
@@ -31,22 +31,41 @@ class Bootloader
       class Config
         # If bootloader should stop on boot menu
         attr_accessor :stop_on_boot_menu
+        # bootloader timeout. Only positive numbers are supported and stop_on_boot_menu has
+        # precedence
+        attr_accessor :timeout
+        # as both previous keys are conflicting, remember which one to set or none. It can be empty
+        # and it means export nothing
+        attr_accessor :keys_to_export
 
         def initialize
+          @keys_to_export = []
```

Comment:
Do you need to track the keys in a separate variable? I would have expected them to be `nil` if they were not specified, so yo need whether to export them or not depending on that condition.

Reply:
well, in this case I prefer to have it explicit..in general I try to avoid nils as it cause troubles on DBus even in this case API is just string one with json encoded. Also using nil means I would need to nullify other value when the first one is set.